_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title:  "The Future of Babel"
+author: Henry Zhu
+date:   2016-11-29 11:00:00
+categories: announcements
+share_text: "The Future of Babel"
+---
+
+How we're doing, where we want to go, and how you can help!
+
+> Previous issues: [Babel Roadmap #4130](https://github.com/babel/babel/issues/4130), [6.0 #2168](https://github.com/babel/babel/issues/2168)
+
+- Project Maintenance
+- Current Status: Where We are Now
+- Future: Where We Want to Go!
+
+## Project Maintenance
+
+Maintaining any large open source project could deserve its own article, but Babel itself covers a lot of ground: core packages, plugins, build integrations such as gulp, webpack, etc (many of which we ourselves may not use), and more. We want to be able to support community plugins, presets, and ideas.
+
+Babel has it's place in many tutorials, guides, talks, etc for various frameworks. The project itself has definitely helped me to understand the language better through bug fixes and support questions. We are active on Twitter, Slack, and Github across other projects that use Babel. We have to balance new features, bug fixes, browser support, speed of compilation/generated code, plugin options, ease of use, working with individuals, big companies, libraries, and more.
+
+I believe our main goal is to make it possible to write JavaScript in any environment you support as seamless as possible and through that support the future of JavaScript tooling.
+
+## Current Status
+
+### Maintaining Babel plugins [for each proposal in TC-39](https://github.com/tc39/proposals) starting from stage-0
+
+Users should be able to easily use features before they are actually released in browsers in stage 4.
+
+This is the core of Babel. It had it's start as a JavaScript transpiler for ES6 back when it was still called 6to5.
+
+In short, proposal champions won't be getting much feedback if no one is actually trying them out in actual code. Babel is a fundamentally important step to the TC-39 process given familiarity in the ecosystem, a simple config change in most people's build process rather than through browser flags, and relative ease of implementation vs. native implementation.
+
+This isn't just a one-time thing as we've found out, especially for early stage proposals. The creation of a plugin means the eventual maintenance of that same plugin. Proposals could completely change in syntax or even get dropped. Since TC-39 meets every 2 months, the plugins should be updated as meetings happen so that users can be in sync and involved with the standards process.
```

Comment:
This might be an ok time to mention [this rule of thumb](https://twitter.com/kentcdodds/status/803815749416456196)

> My rule of thumb on how I decide what future features to transpile:
> "Could I reasonably codemod this if it changes?"
> Don't do it otherwise.

Something like that anyway.